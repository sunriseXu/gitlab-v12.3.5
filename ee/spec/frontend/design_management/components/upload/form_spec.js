import { shallowMount } from '@vue/test-utils';
import UploadForm from 'ee/design_management/components/upload/form.vue';

describe('Design management upload form component', () => {
  let wrapper;

  function createComponent(isSaving = false, canUploadDesign = true) {
    wrapper = shallowMount(UploadForm, {
      sync: false,
      propsData: {
        isSaving,
        canUploadDesign,
        projectPath: '',
        issueIid: '',
      },
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders upload design button', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('hides button if cant upload', () => {
    createComponent(false, false);

    expect(wrapper.element).toMatchSnapshot();
  });

  describe('onFileUploadChange', () => {
    it('emits upload event', () => {
      createComponent();

      wrapper.vm.onFileUploadChange('test');

      expect(wrapper.emitted().upload[0]).toEqual(['test']);
    });
  });
});
