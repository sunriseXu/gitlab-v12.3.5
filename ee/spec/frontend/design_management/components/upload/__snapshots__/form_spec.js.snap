// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management upload form component hides button if cant upload 1`] = `
<header
  class="row-content-block border-top-0 p-2 d-flex"
  issueiid=""
  projectpath=""
>
  <div
    class="d-flex justify-content-between align-items-center w-100"
  >
    <designversiondropdown-stub />
     
    <!---->
  </div>
</header>
`;

exports[`Design management upload form component renders upload design button 1`] = `
<header
  class="row-content-block border-top-0 p-2 d-flex"
  issueiid=""
  projectpath=""
>
  <div
    class="d-flex justify-content-between align-items-center w-100"
  >
    <designversiondropdown-stub />
     
    <uploadbutton-stub />
  </div>
</header>
`;
